lapack/gonum/dgetrf.go

```diff
@@ -28,16 +28,31 @@ import (
 // system of equations.
 func (impl Implementation) Dgetrf(m, n int, a []float64, lda int, ipiv []int) (ok bool) {
 	mn := min(m, n)
-	checkMatrix(m, n, a, lda)
-	if len(ipiv) < mn {
-		panic(badIpiv)
+	switch {
+	case m < 0:
+		panic(mLT0)
+	case n < 0:
+		panic(nLT0)
+	case lda < max(1, n):
+		panic(badLdA)
 	}
-	if m == 0 || n == 0 {
-		return false
```

Comment:
Why was this false?

Reply:
I vaguely remember a small discussion about this a very long time ago. I suppose it was to indicate that no factorization was done but I think that a no-op should not indicate failure.

Reply:
Yes, I agree. This is better.